tests/test_api.py

```diff
@@ -1301,3 +1301,9 @@ def test_no_overwrite_load(tmp_path: Path) -> None:
     with contextlib.suppress(NotImplementedError):
         zarr.load(store)
     assert existing_fpath.exists()
+
+
+def test_auto_chunks() -> None:
+    # Check chunks are automatically set to a sensible default
+    a = zarr.ones(shape=(1000, 1000), dtype=np.uint8)
+    assert a.chunks == (500, 500)
```

Comment:
this test will break if we make small changes the auto-chunking behavior, so maybe it's more future-proof to check that the resulting chunks are the ones you would get by feeding the array parameters into the auto chunk routine.

Reply:
that's reasonable, and the change I'm suggesting can always be made later